File: Algorithm/Implementation/Viral Advertising/Viral Advertising.c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

int viralAdvertising(int n) {
    // Complete this function
    int likes = 2, total_likes = 2;
    for(int i = 1; i < n; i++){
        likes = ( likes * 3) / 2;
        total_likes += likes;
    }
    return total_likes;
}

int main() {
    int n; 
    scanf("%i", &n);
    int result = viralAdvertising(n);
    printf("%d\n", result);
    return 0;
}

File: Algorithm/Implementation/Viral Advertising/Viral Advertising.py
#!/bin/python3

import sys

def viralAdvertising(n):
    # Complete this function
    likes, totalLikes = 2, 2
    for _ in range( n - 1):
        #increment
        likes = ( likes * 3) // 2
        totalLikes += likes
    return totalLikes
    

if __name__ == "__main__":
    n = int(input().strip())
    result = viralAdvertising(n)
    print(result)
